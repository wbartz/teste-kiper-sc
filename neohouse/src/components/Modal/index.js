import PropTypes from 'prop-types';
import React from 'react';
import './index.scss';

const Modal = ({ title, className, children, show }) => {
  return (
    <>
      <div className={`custom-modal modal-overlay ${show ? 'active' : ''}`} />
      <div className={`modal custom-modal ${className}`}>
        <div className="modal-content">
          <h4>{title}</h4>
          <div className="content">{children}</div>
        </div>
      </div>
    </>
  );
};

Modal.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
  show: PropTypes.bool,
  className: PropTypes.string,
};

Modal.defaultProps = {
  title: 'Ops!',
  children: null,
  show: true,
  className: '',
};

export default Modal;
