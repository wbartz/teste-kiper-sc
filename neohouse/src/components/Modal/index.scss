@import '../../styles/variables';

.custom-modal.modal-overlay {
  display: none;
}

.custom-modal.modal-overlay.active {
  z-index: 2;
  display: block;
  opacity: 0.5;
  background-color: $color-black;
  position: fixed;
}

.custom-modal.modal-overlay.active + .modal.custom-modal {
  position: absolute;
  z-index: 3;
  display: block;
  opacity: 1;
  top: 20%;
  transform: scaleX(1) scaleY(1);
}

.modal.custom-modal {
  width: 640px;
  padding: 20px;

  .modal-content {
    h4 {
      text-align: center;
      font-family: $font-family;
      font-size: 45px;
      font-weight: bold;
      color: $color-primary;
    }

    .content {
      font-family: $font-family-alternative;
      font-size: $h5-font-size;
      line-height: 1.5;
      letter-spacing: 0.15px;
      text-align: center;
      margin-top: 20px;
      color: $color-black;
    }
  }
}
