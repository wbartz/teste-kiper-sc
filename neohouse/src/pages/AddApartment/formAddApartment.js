import PropTypes from 'prop-types';
import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import Form from './form';
import * as yup from 'yup';
import Success from './success';
import './index.scss';

const validationSchema = yup.object().shape({
  number: yup.number().required('Este campo é obrigatório'),
});

export const FormAddApartment = ({ history, location, addApartment }) => {
  const [showSuccess, setShowSuccess] = useState(false);
  const { handleSubmit, control, errors } = useForm({
    validationSchema,
  });

  const { block_id } = location.state;

  const onSuccess = () => {
    setShowSuccess(true);
  };

  const handleAdd = (values) => {
    addApartment({ ...values, block_id }, onSuccess);
  };

  return (
    <div className="page add-apartment-page">
      <form onSubmit={handleSubmit(handleAdd)}>
        <Form errors={errors} control={control} />

        <div className="row buttons">
          <div className="col m8">
            <div className="col m2 lef">
              <button
                type="button"
                className="btn btn-flat"
                disabled={errors.length}
                onClick={() => history.goBack()}
              >
                Cancelar
              </button>
            </div>
            <div className="col m2 right">
              <button
                type="submit"
                className="btn btn-submit primary"
                disabled={errors.length}
              >
                Salvar
              </button>
            </div>
          </div>
        </div>
      </form>

      <Success
        show={showSuccess}
        onClose={() => {
          setShowSuccess(false);
          history.push(`${location.pathname.replace('/novo-apartamento', '')}`, {
            block_id: location.state.block_id,
          });
        }}
      />
    </div>
  );
};

FormAddApartment.propTypes = {
  history: PropTypes.shape({ push: PropTypes.func }).isRequired,
  location: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
  addApartment: PropTypes.func.isRequired,
};

export default FormAddApartment;