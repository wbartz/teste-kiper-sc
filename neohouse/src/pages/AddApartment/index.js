import React, { useContext } from 'react';
import { AppContext } from '../../containers/StoreProvider';
import { withRouter } from 'react-router-dom';
import FormAddApartment from './formAddApartment';

export default withRouter((props) => (
  <FormAddApartment {...useContext(AppContext)} {...props} />
));
