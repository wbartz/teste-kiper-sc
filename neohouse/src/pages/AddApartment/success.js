import PropTypes from 'prop-types';
import React from 'react';
import Modal from '../../components/Modal';

const SuccessComponent = ({ show, onClose }) => {
  const handleFinish = () => onClose();

  return (
    <Modal className="edit-modal" show={show} title="Pronto!">
      <div className="row">
        <span>
          O Apartamento foi cadastrado com sucesso!
        </span>
      </div>
      <div className="row buttons">
        <button type="button" className="btn primary" onClick={handleFinish}>
          finalizar
        </button>
      </div>
    </Modal>
  );
};

SuccessComponent.propTypes = {
  show: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default SuccessComponent;
