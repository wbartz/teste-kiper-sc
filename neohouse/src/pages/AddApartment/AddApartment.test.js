import { shallow } from 'enzyme';
import React from 'react';
import Modal from '../../../components/Modal';
import AddApartment from '.';

describe('AddApartment', () => {
  let wrapper;
  const props = {
    history: {
      go: jest.fn(),
    },
    onClose: jest.fn(),
    user: {},
    show: false,
    addApartment: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<AddApartment {...props} />);
  });

  it('render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('render Modal', () => {
    expect(wrapper.find(Modal).exists()).toBeTruthy();
  });
});
