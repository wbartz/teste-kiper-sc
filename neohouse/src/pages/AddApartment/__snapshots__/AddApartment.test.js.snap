// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAddApartment render without crashing 1`] = `ShallowWrapper {}`;
